apps/web/src/app/search-bar.tsx

```diff
@@ -1,34 +1,58 @@
 "use client";
 import { Search } from "lucide-react";
-import { useState } from "react";
+import { useEffect, useState } from "react";
 import { useRouter } from "next/navigation";
 
 export default function SearchBar() {
 	const [searchValue, setSearchValue] = useState("");
+	const [isLoading, setLoading] = useState(false);
+
 	const router = useRouter();
 
-	function triggerSearch() {
+	useEffect(() => {
+		// console.log("Am I loading? " + isLoading);
+	}, [isLoading]);
+
+	// function wait(ms: number) {
+	// 	return new Promise(resolve=>setTimeout(resolve,ms));
+	// }
+
+	async function triggerSearch() {
 		const sv = searchValue.trim();
 		if (sv.length > 0) {
-			router.push(`/search?q=${encodeURIComponent(sv)}`);
+			setLoading(true);
+			await router.push(`/search?q=${encodeURIComponent(sv)}`);
```

Comment:
- This function does not need to be async-
- `router.push` can have the await removed. 
- You don't need to set loading to false bc the next thing that will happen is a page change and that state is not remembered between pages